texture-font.c

```diff
@@ -899,12 +892,30 @@ texture_font_load_glyph( texture_font_t * self,
 
     unsigned char *dst_ptr = buffer + (padding.top * tgt_w + padding.left) * self->atlas->depth;
     unsigned char *src_ptr = ft_bitmap.buffer;
-    for( i = 0; i < src_h; i++ )
-    {
-        //difference between width and pitch: https://www.freetype.org/freetype2/docs/reference/ft2-basic_types.html#FT_Bitmap
-        memcpy( dst_ptr, src_ptr, ft_bitmap.width << (self->atlas->depth == 4 ? 2 : 0));
-        dst_ptr += tgt_w * self->atlas->depth;
-        src_ptr += ft_bitmap.pitch;
+    if( self->atlas->depth == 4 ) {
+        for( i = 0; i < src_h; i++ ) {
+            int j;
+            // flip bgra to rgba, because that's better for OpenGL
+            for( j = 0; j < ft_bitmap.width; j++ ) {
+                uint32_t bgra, rgba;
+                bgra = ((uint32_t*)src_ptr)[j];
+#if __BYTE_ORDER == __BIG_ENDIAN
```

Comment:
See my PR to your repo - __BYTE_ORDER and __BIG_ENDIAN are not defined macros in GCC, so this doesn't work on at least iOS.

Reply:
__BYTE_ORDER is defined in <endian.h>, on Mac OS and iOS, this is <machine/endian.h>. I already check for

    #ifdef __APPLE__
    # include <machine/endian.h>

and this should do it.

Reply:
Ok, looking at your suggested change, it's more complicated.  The people at Apple not only managed to put endian.h into a different location, they also managed to rename the macros.  Doh!  Your fix breaks Linux.

Solution is to first load Apple's endian.h and then redefine the Apple macros to match the Linux macros.

Reply:
Ah! Those tricksy buggers!